### PR TITLE
add D'CENT Wallet support

### PR DESCRIPTION
**Pull request by AnHyoChul**
**Pull Request Summary**

> Describe what changes this pull request makes to the repository

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**

- Add D'CENT Wallet support

**Features**

- Currently, only Astar Network is supported (Shiden Network and Testnet are not supported).

- Staging preview URL
[astar-portal-dcent.vercel.app](https://astar-portal-dcent.vercel.app/)

- Version information for each product that supports Astar Network.
  - D'CENT Biometric Wallet: v2.30.2 or higher
  - D'CENT Wallet for Android: v6.0.4 or higher
  - D'CENT Wallet for iOS: v6.0.6 or higher

### CODE REVIEW

**Comment by AnHyoChul** [comment]
**D'CENT User Guide**
1. Install the D'CENT Wallet on mobile.
2. From the My Wallet tab, click the "Add account+ " button to create an Astar(or Astar-EVM) account.
add_account

3. By clicking the "network" icon on the right top corner of "discovery" tab, you can switch Astar(or Astar-EVM) Network for dapp browser.
Switchnetwork-eng

4. Enter the URL in the address bar.

File: src/config/wallets.ts [hunk 1]
```diff
@@ -292,6 +306,16 @@ export const supportEvmWalletObj = {
...
   },
+  [SupportWallet.DcentEvm]: {
+    img: require('/src/assets/img/icon_dcent.png'),
+    name: "D'CENT Wallet (EVM)",
+    source: SupportWallet.DcentEvm,
+    walletUrl: 'https://store.dcentwallet.com/pages/dcent-wallet-app',
+    guideUrl: 'https://userguide.dcentwallet.com',
+    isSupportBrowserExtension: false,
+    isSupportMobileApp: true,
+    ethExtension: 'ethereum',
```
Comment (gluneau) [inline src/config/wallets.ts:317]: You do not have your own window wallet extension name?

Comment (AnHyoChul) [reply src/config/wallets.ts:317]: @gluneau 
The app uses ethereum.isDcentWallet to distinguish.
And due to the characteristics of our wallet, dApp is supported within the app, so I don't think there is a need to worry about conflicts with other wallets.

**Comment by Kahonnohak** [comment]
Hi!

sorry it took while to get this testing, I have done. Creating account was fine but connecting Portal had problem that I could not find Astar 
Screenshot 2024-02-29 at 11 43 57

**Comment by AnHyoChul** [comment]
@Kahonnohak 
It seems that a few things need to be checked for smooth testing.
Please check the items below.
1. Check the App Information of D'CENT wallet in the Setting menu.
  - D'CENT Wallet for Android: v6.0.4 or higher
  - D'CENT Wallet for iOS: v6.0.6 or higher
Setting

2. Wallet mode must be Biometric or App Wallet.
  -  D'CENT Biometric Wallet: v2.30.2 or higher (if using)
3. When creating an account, ‘Astar’ is the Native wallet and ‘Astar-EVM’ is the EVM wallet.
Account

I hope you will check the above items and give me feedback.

**Comment by gtg7784** [comment]
@AnHyoChul 

Hi D'CENT Team, is there any plans to support Astar zkEVM?

**Comment by AnHyoChul** [comment]
@gtg7784 
We have plans to support Astar zkEVM, but it will proceed after the integration with other chains scheduled in the backlog is completed.

**Comment by gtg7784** [comment]
FYI, https://store.dcentwallet.com/blogs/post/dcent-wallet-expands-astar-network-support-with-zkevm-integration

They support Astar zkEVM too

**Review by gtg7784 — APPROVED** [review]

**Comment by Kahonnohak** [comment]
Now I can see networks, but would this be possible to have only Astar options when access was from portal? 
8D942E69-5FB2-44AC-994F-A8C05519AF47_1_102_o

**Review by impelcrypto — APPROVED** [review]

**Comment by AnHyoChul** [comment]
@Kahonnohak 
After connecting to the portal, you can save the portal URL and network to your favorites by clicking the bookmark mark at the bottom. After that, you can easily connect through Discovery's Favorites. In addition, when this PR is merged, we plan to add Astar Portal to our app's Discovery LIST.
File

**Review by Kahonnohak — APPROVED** [review]
LGTM

**Pull request merged**